simde/simde-complex.h

```diff
@@ -45,12 +45,13 @@
       HEDLEY_HAS_BUILTIN(__builtin_creal) || \
       HEDLEY_GCC_VERSION_CHECK(4,7,0) || \
       HEDLEY_INTEL_VERSION_CHECK(13,0,0) \
-    ) && (!defined(__cplusplus) || !defined(__STRICT_ANSI__))
+    ) && (!defined(__e2k__)) && (!defined(__cplusplus) || !defined(__STRICT_ANSI__))
```

Comment:
I believe this is no longer necessary; with Hedley 15, `HEDLEY_GCC_VERSION_CHECK` will no longer return true for MCST's LCC.

FWIW, I'd be willing to add some logic so we use iF instead of j on LCC. I seem to remember there were some problems with C++, whereas j worked consistently and only gcc-style compilers implement the __builtin_* functions.  That said, I don't think there is much need to have this path taken by lcc when the complex.h version will work fine… this is really only for cases where we don't have a complex.h (like some WebAssembly targets).

Reply:
> I believe this is no longer necessary; with Hedley 15, HEDLEY_GCC_VERSION_CHECK will no longer return true for MCST's LCC.

Hm, won't it break some places, where LCC tends to mimic GCC?

(And yes, then this place won't be needed anymore in such case).

Reply:
Conceivably, yes.  People can still use `HEDLEY_GNUC_VERSION_CHECK` if they really want to trust `__GNUC__`/`__GNUC_MINOR__`/`__GNUC_PATCHLEVEL__`, but IMHO that's generally a bad idea.  LCC actually does a pretty good job of supporting GNU extensions compared to a lot of compilers, but even it fails for lots of stuff we want to check… like this code.  There are also the pragmas to control diagnostics (LCC uses the EDG syntax instead of the GCC syntax), pushing/popping the warning stack, how _Noreturn works, etc.

I think it's fair to argue that defining `__GNUC__` and friends is a good decision in general for LCC (and other compilers which do it), but when we actually have access to the compiler and can test we can be much more accurate by separating them.  It's also important to note that failures from trusting `__GNUC__` tend to break the code, whereas if you test for LCC separately failures just mean you don't get that feature.

If you take a look at [the patch which added LCC support to Hedley](https://github.com/nemequ/hedley/commit/099ed01325d83b36a5154c3ace6cad9a87d03bf1), you'll see that a lot of tests could be eliminated by using `HEDLEY_GNUC_VERSION_CHECK` instead of `HEDLEY_GCC_VERSION_CHECK`, but there are also a bunch where we have to do something for LCC because the GCC syntax isn't supported.

Reply:
Ok. Seems to do well now, once I rebased this PR to `wip/e2k`; I found some problem BTW related to LCC mimics GCC no more, and fixed it in 0b4ee01.